packages/codegen-ui-react/lib/react-studio-template-renderer.ts

```diff
@@ -1296,5 +1302,52 @@ export abstract class ReactStudioTemplateRenderer extends StudioTemplateRenderer
     return undefined;
   }
 
+  /* To map predefined props to children props for variants
+   *
+   * Example: Text prop label is mapped to to Text prop Children
+   *
+   */
+  private mapSyntheticPropsForVariants() {
+    if (!isStudioComponentWithVariants(this.component)) {
+      return;
+    }
+
+    this.component.variants.forEach((variant) => {
+      // loop through the keys in the dict. Ex. Button, Flex.Flex[0].Flex[0].Button[0]
+      Object.entries(variant.overrides).forEach(([overrideKey, value]) => {
+        const propsInOverrides = value;
+        const componentType = this.getComponentTypeFromOverrideKey(overrideKey);
+        if (isPrimitive(componentType)) {
+          const childrenPropMapping = PrimitiveChildrenPropMapping[Primitive[componentType as Primitive]];
+          if (childrenPropMapping !== undefined) {
+            // only remap if children prop is not defined in this particular overrides section
+            if (propsInOverrides.children === undefined && propsInOverrides[childrenPropMapping] !== undefined) {
+              propsInOverrides.children = propsInOverrides[childrenPropMapping];
+              delete propsInOverrides[childrenPropMapping];
+            }
+          }
+        }
+      });
+    });
+  }
+
+  /* Parse the override key to get the component type
+   *
+   * Example1: Flex.Flex[0].Flex[0].Button[0]
+   * Example2: Button
+   *
+   */
+  private getComponentTypeFromOverrideKey(overrideKey: string): string {
```

Comment:
Could you move this method to the [StudioNode](https://github.com/aws-amplify/amplify-codegen-ui/blob/main/packages/codegen-ui/lib/studio-node.ts) class as a `static` method, and ensure we've got unit tests covering it?